core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -1324,87 +1324,9 @@ public synchronized Info fetchDockerDaemonInfo() throws IOException {
-        File sourceFile = new File(mountableFile.getResolvedPath());
-
-        if (containerPath.endsWith("/") && sourceFile.isFile()) {
-            logger().warn("folder-like containerPath in copyFileToContainer is deprecated, please explicitly specify a file path");
-            copyFileToContainer((Transferable) mountableFile, containerPath + sourceFile.getName());
-        } else {
-            copyFileToContainer((Transferable) mountableFile, containerPath);
-        }
-    }
-
-    @Override
-    @SneakyThrows(IOException.class)
-    public void copyFileToContainer(Transferable transferable, String containerPath) {
-        if (!isCreated()) {
-            throw new IllegalStateException("copyFileToContainer can only be used with created / running container");
-        }
-
-        try (
-            ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
-            TarArchiveOutputStream tarArchive = new TarArchiveOutputStream(byteArrayOutputStream)
-        ) {
-            tarArchive.setLongFileMode(TarArchiveOutputStream.LONGFILE_POSIX);
-
-            transferable.transferTo(tarArchive, containerPath);
-            tarArchive.finish();
-
-            dockerClient
-                .copyArchiveToContainerCmd(containerId)
-                .withTarInputStream(new ByteArrayInputStream(byteArrayOutputStream.toByteArray()))
-                .withRemotePath("/")
-                .exec();
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
+    @SneakyThrows
     public void copyFileFromContainer(String containerPath, String destinationPath) {
-        copyFileFromContainer(containerPath, inputStream -> {
-            try(FileOutputStream output = new FileOutputStream(destinationPath)) {
-                IOUtils.copy(inputStream, output);
-                return null;
-            }
-        });
-    }
-
-    @Override
-    @SneakyThrows(Exception.class)
-    public <T> T copyFileFromContainer(String containerPath, ThrowingFunction<InputStream, T> consumer) {
-        if (!isCreated()) {
-            throw new IllegalStateException("copyFileFromContainer can only be used when the Container is created.");
-        }
-
-        try (
-            InputStream inputStream = dockerClient.copyArchiveFromContainerCmd(containerId, containerPath).exec();
-            TarArchiveInputStream tarInputStream = new TarArchiveInputStream(inputStream)
-        ) {
-            tarInputStream.getNextTarEntry();
-            return consumer.apply(tarInputStream);
-        }
-    }
-
-    /**
-     * {@inheritDoc}
-     */
-    @Override
-    public ExecResult execInContainer(Charset outputCharset, String... command)
-            throws UnsupportedOperationException, IOException, InterruptedException {
-        return ExecInContainerPattern.execInContainer(getContainerInfo(), outputCharset, command);
+        Container.super.copyFileFromContainer(containerPath, destinationPath);
```

Comment:
because it was defined as `throws IOException, InterruptedException` in `Container`